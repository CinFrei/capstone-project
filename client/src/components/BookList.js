import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

import BookCard from './BookCard'

/** @component */
export default BookList

BookList.propTypes = {
  listName: PropTypes.string,
  newBooks: PropTypes.array,
  showDetail: PropTypes.func,
}

function BookList({ listName, newBooks, showDetail }) {
  return (
    <BookListWrapper>
      <h2>{listName}</h2>
      {newBooks.map((newBook) => (
        <BookCard
          author={newBook.volumeInfo.authors}
          buttonName="Mehr Ã¼ber's Buch"
          cover={newBook.volumeInfo.imageLinks.thumbnail}
          onCardButtonClick={() => showDetail(newBook.id)}
          key={newBook.id}
          title={newBook.volumeInfo.title}
        />
      ))}
    </BookListWrapper>
  )
}

const BookListWrapper = styled.ol`
  display: grid;
  grid-gap: 10px;
  grid-template-rows: 1fr auto;
  padding: 20px;

  h2 {
    margin: 0;
    text-align: left;
  }
`
