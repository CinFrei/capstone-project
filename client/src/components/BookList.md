Basic BookList:

```js
<BookList
newBooks={{
    books: {
    1: { author: '["Paulchen Panter", "Peter Pan"]',
      buttonName: 'Buch entfernen',
      cover: './public/styleguide/sample-cover.jpg"',
      title: 'Pauls Rosa Welt',},
    2: { author: '["Paulchen Panter", "Peter Pan"]',
      buttonName: 'Buch entfernen',
      cover: './public/styleguide/sample-cover.jpg"',
      title: 'Pauls Rosa Welt',},
    },
  }
  deleteBook={() => {}}
  listName="BÃ¼cherregal"

/>
```
