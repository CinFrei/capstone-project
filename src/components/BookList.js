import BookCard from './BookCard'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

/** @component */
export default BookList

BookList.propTypes = {
  listName: PropTypes.string,
  newBooks: PropTypes.array,
}

function BookList({ newBooks, listName, deleteBook }) {
  return (
    <BookListWrapper>
      <h2>{listName}</h2>
      {newBooks.map((newBook) => (
        <BookCard
          author={newBook.volumeInfo.authors}
          buttonName="Buch entfernen."
          cover={newBook.volumeInfo.imageLinks.thumbnail}
          key={newBook.id}
          onCardBtnClick={() => deleteBook(newBook.id)}
          title={newBook.volumeInfo.title}
        />
      ))}
    </BookListWrapper>
  )
}

const BookListWrapper = styled.section`
  display: grid;
  grid-gap: 10px;
  grid-template-rows: 1fr auto;
  padding: 10px;

  h2 {
    margin: 0;
    text-align: left;
  }
`
