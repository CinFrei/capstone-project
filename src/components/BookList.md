Basic BookList:

```js
<BookList listName="BÃ¼cherregal" />
```