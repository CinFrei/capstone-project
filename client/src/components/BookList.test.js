import BookList from './BookList'
import { render } from '@testing-library/react'

describe('BookList', () => {
  it('renders correctly', () => {
    const deleteBookMock = jest.fn()
    const { container, getByText, getByRole } = render(
      <BookList
        deleteBook={deleteBookMock}
        listName="BÃ¼cheregal"
        newBooks={[
          {
            volumeInfo: {
              authors: [`Paul Tobin`],
              imageLinks: {
                thumbnail:
                  'http://books.google.com/books/content?id=6D13DwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
              },
              title: 'The Witcher, Band 2 - Fuchskinder',
            },
          },
        ]}
      />
    )

    expect(container.firstChild).toMatchSnapshot()

    expect(getByText('Paul Tobin')).toBeInTheDocument()
    expect(getByRole('img')).toBeInTheDocument()
    expect(getByText('The Witcher, Band 2 - Fuchskinder')).toBeInTheDocument()
  })
})
