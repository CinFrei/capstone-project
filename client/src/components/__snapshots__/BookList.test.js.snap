// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList renders correctly 1`] = `
.c3 {
  background: var(--accent-color);
  border-radius: 3px;
  border: none;
  color: var(--primary-anthrazit);
  font-family: 'Kaushan Script',cursive;
  font-size: 1.2em;
  grid-column: 2 / -1;
  justify-self: end;
  width: 100%;
}

.c1 {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 40% 60%;
  grid-template-rows: 1fr 45px;
  min-width: 0;
  padding: 10px;
  position: relative;
}

.c1 h3 {
  margin: 0;
}

.c1 h4 {
  display: inline;
  margin: 10px;
}

.c1 span {
  margin: 0 0 0 3.5px;
}

.c1 p {
  margin: 15px 5px 0 3px;
}

.c2 {
  grid-column: 1 / span 1;
  grid-row: 1 / span 2;
  justify-self: stretch;
}

.c0 {
  display: grid;
  grid-gap: 10px;
  grid-template-rows: 1fr auto;
  padding: 10px;
}

.c0 h2 {
  margin: 0;
  text-align: left;
}

<ol
  class="c0"
>
  <h2>
    Bücheregal
  </h2>
  <li
    class="c1"
  >
    <img
      alt="The Witcher, Band 2 - Fuchskinder"
      class="c2"
      src="http://books.google.com/books/content?id=6D13DwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
    />
    <div>
      <h3>
        The Witcher, Band 2 - Fuchskinder
      </h3>
      <span>
        von
        <h4>
          Paul Tobin
        </h4>
      </span>
    </div>
    <button
      class="c3"
    >
      Buch entfernen.
    </button>
    <button
      class="c3"
    >
      Mehr über's Buch
    </button>
  </li>
</ol>
`;
